File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/RoomDatabaseDelegate.java
package org.coralibre.android.sdk.internal.database.persistent;

import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import org.coralibre.android.sdk.internal.database.persistent.entity.EntityCapturedData;
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityDiagnosisKey;
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityExposureInformation;
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityExposureSummary;
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityTemporaryExposureKey;
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityToken;


@androidx.room.Database(
        entities = {
            EntityTemporaryExposureKey.class,
            EntityCapturedData.class,
            EntityToken.class,
            EntityDiagnosisKey.class,
            EntityExposureInformation.class,
            EntityExposureSummary.class
        },
        version = 1,
        exportSchema = false
)
@TypeConverters(
        {ENIntervalConverter.class}
)
public abstract class RoomDatabaseDelegate extends RoomDatabase {

    public abstract DaoTEK daoTEK();
    public abstract DaoCapturedData daoCapturedData();
    public abstract DaoToken daoToken();
    public abstract DaoDiagnosisKey daoDiagnosisKey();
    public abstract DaoExposureInformation daoExposureInformation();
    public abstract DaoExposureSummary daoExposureSummary();

}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/RoomDatabaseDelegate.kt
package org.coralibre.android.sdk.internal.database.persistent

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityCapturedData
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityDiagnosisKey
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityExposureInformation
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityExposureSummary
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityTemporaryExposureKey
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityToken

@Database(
    entities = [
        EntityCapturedData::class,
        EntityDiagnosisKey::class,
        EntityExposureInformation::class,
        EntityExposureSummary::class,
        EntityTemporaryExposureKey::class,
        EntityToken::class,
    ],
    version = 1,
    exportSchema = false
)
@TypeConverters(
    ENIntervalConverter::class
)
abstract class RoomDatabaseDelegate : RoomDatabase() {
    abstract fun daoTEK(): DaoTEK
    abstract fun daoCapturedData(): DaoCapturedData
    abstract fun daoToken(): DaoToken
    abstract fun daoDiagnosisKey(): DaoDiagnosisKey
    abstract fun daoExposureInformation(): DaoExposureInformation
    abstract fun daoExposureSummary(): DaoExposureSummary
}
